Zarhus

m-iwanicki opened the pull request:
```sh
robot -L TRACE -b cmd_logs.txt -v rte_ip:127.0.0.1 -v config:qemu -v snipeit:no \
    -v boot_dts_from_ipxe_shell:True \
    -v dts_ipxe_link:https://boot.dasharo.com/dts/dts-rc.ipxe \
    -v features:encryption,otab \
    -v zarhus_wic_gz_file:../yocto/zarhus/zarhus-base-image-genericx86-64.rootfs.wic.gz \
    -v zarhus_swu_file:../yocto/zarhus/zarhus-swu-image-genericx86-64.rootfs.swu \
    zarhus-x86/zarhus.robot
```

Results:

```
==============================================================================
Zarhus
==============================================================================
ZHS001.206 Make sure that cukinia tests pass                          | PASS |
------------------------------------------------------------------------------
ZHS002.206 Make sure that update works :: Make sure that update vi... | PASS |
------------------------------------------------------------------------------
ZHS003.206 Make sure that failed update rollbacks to working slot ... | PASS |
------------------------------------------------------------------------------
ZHS004.206 Make sure that rollback works :: Make sure that `otab r... | PASS |
------------------------------------------------------------------------------
Zarhus                                                                | PASS |
4 tests, 4 passed, 0 failed
==============================================================================
```

[log.html](https://github.com/user-attachments/files/22901313/log.html)

On hardware:

```sh
robot -L TRACE -b cmd_logs.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no \
    -v boot_dts_from_ipxe_shell:True \
    -v dts_ipxe_link:https://boot.dasharo.com/dts/dts-rc.ipxe \
    -v features:encryption,otab \
    -v zarhus_wic_gz_file:../yocto/zarhus/zarhus-base-image-genericx86-64.rootfs.wic.gz \
    -v target_device:nvme0n1 -t "ZHS001*" zarhus-x86/zarhus.robot
```

```
==============================================================================
Zarhus
==============================================================================
ZHS001.206 Make sure that cukinia tests pass                          | PASS |
------------------------------------------------------------------------------
Zarhus                                                                | PASS |
1 test, 1 passed, 0 failed
==============================================================================
```

[log.html](https://github.com/user-attachments/files/22908373/log.html)

---

```sh
robot -L TRACE -b command_log.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no \
    -v boot_dts_from_ipxe_shell:True \
    -v dts_ipxe_link:https://boot.dasharo.com/dts/dts-rc.ipxe \
    -v features:encryption,otab \
    -v zarhus_bootstrap_file:../yocto/zarhus-dtrpb/bootstrap.img \
    zarhus-x86/zarhus-provisioning-preparation.robot
```

```
==============================================================================
Zarhus-Provisioning-Preparation
==============================================================================
ZPBP001.101 Make sure that network boot is disabled :: Make sure t... | PASS |
------------------------------------------------------------------------------
ZPBP002.101 Make sure that BIOS boot medium is locked :: Make sure... | PASS |
------------------------------------------------------------------------------
ZPBP003.101 Make sure that SMM BIOS write protection is enabled ::... | PASS |
------------------------------------------------------------------------------
ZPBP004.101 Make sure that Intel ME is HAP disabled :: Make sure t... | PASS |
------------------------------------------------------------------------------
ZPBP005.101 Make sure that OptionROM loading is disabled :: Make s... | PASS |
------------------------------------------------------------------------------
ZPBP006.206 Make sure that zarhus status returns expected initial ... | PASS |
------------------------------------------------------------------------------
ZPBP006.206 Make sure that Secure Boot is in Setup Mode :: Make su... | FAIL |
'Installed:      ✗ sbctl is not installed
Setup Mode:     ✗ Enabled
Secure Boot:    ✗ Disabled
Vendor Keys:    none' does not match 'Setup Mode:*Enabled'
------------------------------------------------------------------------------
Zarhus-Provisioning-Preparation                                       | FAIL |
7 tests, 6 passed, 1 failed
==============================================================================
```

After fix:

```
==============================================================================
ZPBP006.206 Make sure that zarhus status returns expected initial ... | PASS |
------------------------------------------------------------------------------
ZPBP006.206 Make sure that Secure Boot is in Setup Mode :: Make su... | PASS |
Secure Boot is in Setup Mode
------------------------------------------------------------------------------
Zarhus-Provisioning-Preparation                                       | PASS |
2 tests, 2 passed, 0 failed
==============================================================================
```

And second fix later to change `ZPBP006` -> `ZPBP007`

Couldn't test `zarhus-provisioning.robot` as Suite preparation fails in `Setup ZPB` keyword, `Should Contain    ${status}    Boot Guard: Enabled` to be exact. For some reason capsule update doesn't work.

On QEMU:

```
==============================================================================
Zarhus-Provisioning
==============================================================================
ZPB001.206 Make sure that cukinia tests pass                          | PASS |
------------------------------------------------------------------------------
Zarhus-Provisioning                                                   | PASS |
1 test, 1 passed, 0 failed
==============================================================================
```

m-iwanicki committed:
lib: QemuMonitor: Allow adding removable and writable USB

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: QemuMonitor: Add keyword to remove added USB

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts-lib: move checkpoint keywords to separate lib so it can be used for Zarhus

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
zarhus-lib: Add basic keywords for OS preparation and credentials

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: network: make `Send File To DUT` work on OS without root access

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: terminal: Add `Execute Command In Terminal Should Succeed` keyword

Used to run commands and check if they succeeded

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
zarhus-x86: zarhus.robot: add cukinia test and couple update tests

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
docs: zarhus-tests: Add documentation related to Zarhus OS testing

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: network: add 'Send File To DUT Directly' keyword

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: menus.py: Add 'OptionROM Execution Policy' option

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: options-lib_uefi-setup-menu: fix Get UEFI Option Documentation

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: usb-hid-msc-lib: add keyword to return first USB stick

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
lib: zarhus-provision-lib: Add keywords to use with ZPB

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

iwanicki92 commented on lib/zarhus-lib.robot:
```diff
@@ -0,0 +1,208 @@
+    ...    - First boot setup done (set encryption password, user password changed)
+    ...
+    ...    === Effects ===
+    ...    - Power platform on
```
```suggestion
```

iwanicki92 commented on lib/zarhus-lib.robot:
```diff
@@ -0,0 +1,208 @@
+
+Install Zarhus OS
+    [Documentation]    Prepare Zarhus OS for testing. Flash it to
+    ...    ${TARGET_DEVICE}, or /dev/sda if testing on QEMU and add to
```
```suggestion
    ...    ${TARGET_DEVICE}, or /dev/sda if testing on QEMU and add it
    ...    as a default boot option
```

iwanicki92 commented on docs/zarhus-tests.md:
````diff
@@ -0,0 +1,223 @@
+```sh
+robot -L TRACE -b cmd_logs.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no -v boot_dts_from_ipxe_shell:True \
+    -v dts_ipxe_link:https://boot.dasharo.com/dts/dts-rc.ipxe \
+    -v zarhus_bootstrap_file:../yocto/zarhus-dtrpb/bootstrap.img \
+    zarhus-x86/zarhus-provisioning-preparation.robot
````
```suggestion
    -v features:encryption,otab \
    -v zarhus_bootstrap_file:../yocto/zarhus-dtrpb/bootstrap.img \
    zarhus-x86/zarhus-provisioning-preparation.robot
```

iwanicki92 commented on lib/zarhus-provision-lib.robot:
```diff
@@ -0,0 +1,121 @@
+    ...    \ OS on ``${disk}`` and flashes pre-configured firmware image
+    ...    - Runs first boot setup of Zarhus OS
+    [Arguments]    ${disk}=nvme0n1
+    Install ZPB OS    ${disk}
```
```suggestion
    Set Zarhus Features
    Install ZPB OS    ${disk}
```

iwanicki92 commented on lib/zarhus-lib.robot:
```diff
@@ -0,0 +1,208 @@
+Set Zarhus Features
+    [Documentation]    Return features in global ``${FEATURES}`` variable as a
+    ...    list. Features should be split with comma. Returns empty list if
+    ...    features is not defined
```
```suggestion
    [Documentation]    Set features from ``${FEATURES}`` variable to
    ...    ``${ZARHUS_FEATURES}`` list with Suite scope. Features should
    ...    be split with comma.
```

iwanicki92 commented on lib/zarhus-provision-lib.robot:
```diff
@@ -0,0 +1,121 @@
+    Enter Setup Menu Tianocore
+
+Prepare ZPB OS
+    [Documentation]    Install ZPB OS via ZPB bootstrap USB
```
```suggestion
    [Documentation]    Install ZPB OS via ZPB bootstrap USB
    ...    and do first boot setup.
```

iwanicki92 commented on zarhus-x86/zarhus-provisioning-preparation.robot:
```diff
@@ -0,0 +1,66 @@
+ZPBP005.101 Make sure that OptionROM loading is disabled
+    [Documentation]    Make sure that Network Boot is disabled
+    ${state}=    Get UEFI Option    OptionROMExecutionPolicy
+    Should Be Equal    ${state}    Disable all OptionROMs
```
```suggestion
    Should Contain    ${state}    Disable all OptionROMs
```

iwanicki92 commented on zarhus-x86/zarhus.robot:
```diff
@@ -0,0 +1,105 @@
+    Should Be Equal    ${current_part}    ${current_part2}
+
+ZHS004.206 Make sure that rollback works
+    [Documentation]    Make sure that `otab rollback` works
```
```suggestion
    [Documentation]    Make sure that `otab rollback` will result in 
    ...    booting into inactive slot after reboot.
```

iwanicki92 commented on os-config/206-credentials.py:
```diff
@@ -0,0 +1,10 @@
+# SPDX-FileCopyrightText: 2025 3mdeb <contact@3mdeb.com>
```
Missing changes to `environment-test-ids.py`, probably lost during rebase.

iwanicki92 commented on lib/usb-hid-msc-lib.robot:
```diff
@@ -74,3 +75,31 @@ Check USB Stick Detection In Edk2
+        # Removable USB storage (size > 0), that is not partition
+        VAR    ${command}=
+        ...    test ! -f "/sys/class/block/${device}/partition"
+        ...    test "\$(cat /sys/class/block/${device}/size)"
```
```suggestion
        ...    test "\$(cat /sys/class/block/${device}/size)" -gt 0
```

iwanicki92 commented on zarhus-x86/zarhus-provisioning-preparation.robot:
```diff
@@ -0,0 +1,66 @@
+Resource            ../lib/zarhus-provision-lib.robot
+
+Suite Setup         Run Keywords
+...                     Prepare Test Suite
+...                     AND
```
```suggestion
...                     Prepare Test Suite
...                     AND
...                     Skip If    '${MANUFACTURER}' == 'QEMU'
...                     AND
```

iwanicki92 commented on lib/zarhus-lib.robot:
```diff
@@ -0,0 +1,208 @@
+    ...    gunzip -c /tmp/zarhus.wic.gz | dd of=/dev/${device}
+    ...    err_msg=Flashing Zarhus OS failed!
+    ...    timeout=5m
+    VAR    ${DEVICE_OS_USERNAME}=    ${os_username_old}    scope=Suite
```
```suggestion
    # Using `scope=Suite` because `scope=Test` resulted in those variables
    # having root/${EMPTY} values in tests (after suite setup finished)
    VAR    ${DEVICE_OS_USERNAME}=    ${os_username_old}    scope=Suite
```

iwanicki92 commented on lib/QemuMonitor.py:
```diff
@@ -181,6 +156,7 @@ def usb_add(self, img_path, name="unnamed"):
             "driver": "usb-storage",
             "id": self._usb_devid(name),  # "usbdisk",
             "drive": self._usb_nodename(name),
+            "removable": "on" if removable else "off",
```
```suggestion
            "removable": removable,
```

iwanicki92 commented on docs/zarhus-tests.md:
````diff
@@ -0,0 +1,223 @@
+Example test run that succeeded:
+
+```sh
+robot -L TRACE -b cmd_logs.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no \
````
```suggestion
robot -L TRACE -b command_log.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no \
```

iwanicki92 commented on docs/zarhus-tests.md:
````diff
@@ -0,0 +1,223 @@
+Example command:
+
+```sh
+robot -L TRACE -b cmd_logs.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no -v boot_dts_from_ipxe_shell:True \
````
```suggestion
robot -L TRACE -b command_log.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no -v boot_dts_from_ipxe_shell:True \
```

iwanicki92 commented on lib/zarhus-provision-lib.robot:
```diff
@@ -0,0 +1,121 @@
+    ...    - Runs first boot setup of Zarhus OS
+    [Arguments]    ${disk}=nvme0n1
+    Install ZPB OS    ${disk}
+    Boot Dasharo Tools Suite    iPXE
+    Enter Shell In DTS
+    # Make sure Zarhus is first boot entry
+    Execute Command In Terminal Should Succeed
+    ...    efibootmgr --disk "/dev/${disk}" --part 1 --create --label "ZarhusOS" --loader '\\EFI\\BOOT\\bootx64.efi'
```
```suggestion
    Execute Command In Terminal Should Succeed
    ...    mount -o remount,rw efivarfs
    ...    Failed to remount efivarfs as read-writable
    Execute Command In Terminal Should Succeed
    ...    efibootmgr --disk "/dev/${disk}" --part 1 --create --label "ZarhusOS" --loader '\\EFI\\BOOT\\bootx64.efi'
```

m-iwanicki committed:
lib/bios/menus.py: SMM BIOS write protection is split in 2 lines

Example on Dasharo v0.9.1 on ODROID-H4+:

```
Enable SMM BIOS write      [X]
protection
```

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
zarhus: zarhus-provisioning: Add tests for preparation and cukinia

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

iwanicki92 commented on lib/zarhus-lib.robot:
```diff
@@ -0,0 +1,211 @@
+    Execute Command In Terminal Should Succeed
+    ...    test -b /dev/${device}    err_msg=/dev/${device} block device doesn't exist!
+    Execute Command In Terminal Should Succeed
+    ...    gunzip -c /tmp/zarhus.wic.gz | dd of=/dev/${device}
```
`set -o pipefail` in case `gunzip` fails

m-iwanicki replied:
[c02c9...ee3d0](https://github.com/Dasharo/open-source-firmware-validation/compare/c02c919707a6257602e7e9ba6273e0b996981485..ee3d0b4b67b711e74c5c889cedf337ea61c7f941)

3mkusiak commented on docs/zarhus-tests.md:
```diff
@@ -0,0 +1,242 @@
+
+## Automatic tests
+
+**Important**: To run OSFV tests on Zarhus OS with encryption feature enabled
```
This should highlight this should be done at building stage. We're mentioning provisioning box here anyway so I think we might add the path (not link) for which file to edit as well. Spares the reproducer time to figure it out.

m-iwanicki replied:
[21147..82184](https://github.com/Dasharo/open-source-firmware-validation/compare/21147742280aa7e4fe02407100e09f4986bb3f90..821849f25a1e6248235c70942eb7516f4f323a18)

3mkusiak commented on docs/zarhus-tests.md:
```diff
@@ -0,0 +1,242 @@
+
+#### Zarhus Provisioning Box
+
+Before starting, flash firmware manually.
```
Please add short explanation why this step needs to be done, there is no context to it. Might as well add link to docs dasharo firmares for convenience, currently odroid is supported anyway.

m-iwanicki replied:
> currently

and possibly not always [21147..82184](https://github.com/Dasharo/open-source-firmware-validation/compare/21147742280aa7e4fe02407100e09f4986bb3f90..821849f25a1e6248235c70942eb7516f4f323a18)

3mkusiak commented on lib/usb-hid-msc-lib.robot:
```diff
@@ -74,3 +75,31 @@ Check USB Stick Detection In Edk2
+        # Removable USB storage (size > 0), that is not partition
+        VAR    ${command}=
+        ...    test ! -f "/sys/class/block/${device}/partition"
+        ...    test "\$(cat /sys/class/block/${device}/size)" -gt 0
```
Very corner case I suppose but specifying size might spare somebody debug in the future.

m-iwanicki replied:
I'm not sure I understand? I verify size (`-gt 0`) is greater than 0

3mkusiak reviewed (commented):
Contentwise looks good to me, small nitpicks for now. I'll proceed to running tests.

3mkusiak commented on docs/zarhus-tests.md:
````diff
@@ -0,0 +1,242 @@
+
+Example command to run tests:
+
+```sh
````
It must be highlighted how to prepare that bootsrap file and how to mount it and edit if needed.

m-iwanicki replied:
[21147..82184](https://github.com/Dasharo/open-source-firmware-validation/compare/21147742280aa7e4fe02407100e09f4986bb3f90..821849f25a1e6248235c70942eb7516f4f323a18)

ghost commented:
Tested provisioning box on hardware, details below

<details>
<summary> HW PROV.BOX </summary>

All tests have passed beside one

```log
(venv) mkusiak in ~/Development/dts/open-source-firmware-validation on zarhus ● λ robot -L TRACE -b command_log.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no -v boot_dts_from_ipxe_shell:True \
    -v dts_ipxe_link:https://boot.dasharo.com/dts/dts-rc.ipxe \
    -v features:encryption,otab \
    -v zarhus_bootstrap_file:"$bootstrap" \
    zarhus-x86/zarhus-provisioning-preparation.robot
==============================================================================
Zarhus-Provisioning-Preparation                                               
==============================================================================
ZPBP001.101 Make sure that network boot is disabled :: Make sure t... | PASS |
------------------------------------------------------------------------------
ZPBP002.101 Make sure that BIOS boot medium is locked :: Make sure... | PASS |
------------------------------------------------------------------------------
ZPBP003.101 Make sure that SMM BIOS write protection is enabled ::... | PASS |
------------------------------------------------------------------------------
ZPBP004.101 Make sure that Intel ME is HAP disabled :: Make sure t... | PASS |
------------------------------------------------------------------------------
ZPBP005.101 Make sure that OptionROM loading is disabled :: Make s... | PASS |
------------------------------------------------------------------------------
ZPBP006.206 Make sure that zarhus status returns expected initial ... | FAIL |
low != high
------------------------------------------------------------------------------
ZPBP007.206 Make sure that Secure Boot is in Setup Mode :: Make su... | PASS |
Secure Boot is in Setup Mode
------------------------------------------------------------------------------
Zarhus-Provisioning-Preparation                                       | FAIL |
7 tests, 6 passed, 1 failed
==============================================================================
Debug:   /home/mkusiak/Development/dts/open-source-firmware-validation/command_log.txt
Output:  /home/mkusiak/Development/dts/open-source-firmware-validation/output.xml
Log:     /home/mkusiak/Development/dts/open-source-firmware-validation/log.html
Report:  /home/mkusiak/Development/dts/open-source-firmware-validation/report.html
```

One of the tests failed due to improper state of power LED, this is a known issue.

<img width="651" height="328" alt="image" src="https://github.com/user-attachments/assets/c746653c-9c72-4b5f-a999-2c951437c9f0" />

</details>

Will proceed to test zarhus os.

3mkusiak commented on docs/zarhus-tests.md:
```diff
@@ -0,0 +1,260 @@
+**Important**: To run OSFV tests on Zarhus OS with encryption feature enabled
+you have to make sure that kernel command line contains `console=ttyS0,115200`
+instead of `console=tty1`, as disk encryption password prompt displayed in
+initramfs shows only on that `tty`. Running tests with wrong image will result
```
This is a bit misleading, as it can be understood as "if encryption layer is added it is necessary to edit CMDLINE".

m-iwanicki replied:
[82184..d10cc](https://github.com/Dasharo/open-source-firmware-validation/compare/821849f25a1e6248235c70942eb7516f4f323a18..d10cc6344d54e1cd8c1e759af00b9206e860632b)

3mkusiak commented on docs/zarhus-tests.md:
````diff
@@ -0,0 +1,260 @@
+Example test run that succeeded:
+
+```sh
+robot -L TRACE -b command_log.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no \
````
Might be wise to list what layers shall be built in. Eg. this won't run without encryption support added.

m-iwanicki replied:
> Might be wise to list what layers shall be built in

The ones you want to test. I mentioned in docs that only `meta-zarhus-hardening` is required.
Tests should be skipped if they are not supported e.g.: https://github.com/Dasharo/open-source-firmware-validation/blob/308dc4a10fdcc7e2973cda23b50c6a3d784a4261/dasharo-compatibility/audio-subsystem.robot#L54-L56

m-iwanicki replied:
> Eg. this won't run without encryption support added.

I'd assume that person testing knows what's he's doing. There is variable `-v features:encryption,otab` which lists features your image supports, described more extensively in `### Test Variables`. Expanded a little to mention which `.yml` files are required for this feature.

[82184..d10cc](https://github.com/Dasharo/open-source-firmware-validation/compare/821849f25a1e6248235c70942eb7516f4f323a18..d10cc6344d54e1cd8c1e759af00b9206e860632b)

m-iwanicki replied:
Also added skip to update tests if no otab feature available: [d10cc..39dae](https://github.com/Dasharo/open-source-firmware-validation/compare/d10cc6344d54e1cd8c1e759af00b9206e860632b..39daed090aef152e0e6547911eb07d0ae048402b):

```
robot -L TRACE -b cmd_logs.txt -v rte_ip:192.168.10.168 -v config:odroid-h4-plus -v snipeit:no -v boot_dts_from_ipxe_shell:True \
    -v dts_ipxe_link:https://boot.dasharo.com/dts/dts-rc.ipxe \
        -v features:encryption \
        -v zarhus_wic_gz_file:../yocto/zarhus/zarhus-base-image-genericx86-64.rootfs.wic.gz \
        -v target_device:nvme0n1 -t "ZHS00[234]*" zarhus-x86/zarhus.robot
==============================================================================
Zarhus
==============================================================================
ZHS002.206 Make sure that update works :: Make sure that update vi... | SKIP |
ZHS002.206 not supported
------------------------------------------------------------------------------
ZHS003.206 Make sure that failed update rollbacks to working slot ... | SKIP |
ZHS003.206 not supported
------------------------------------------------------------------------------
ZHS004.206 Make sure that rollback works :: Make sure that `otab r... | SKIP |
ZHS004.206 not supported
------------------------------------------------------------------------------
Zarhus                                                                | SKIP |
3 tests, 0 passed, 0 failed, 3 skipped
==============================================================================
```

ghost commented:
Hardware ZarhusOS tests pass

<details><summary>ZarhusOS HW</summary>
<p>

```log
==============================================================================
Zarhus                                                                        
==============================================================================
ZHS001.206 Make sure that cukinia tests pass                          | PASS |
------------------------------------------------------------------------------
Zarhus                                                                | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Debug:   /home/mkusiak/Development/dts/open-source-firmware-validation/command_log.txt
Output:  /home/mkusiak/Development/dts/open-source-firmware-validation/output.xml
Log:     /home/mkusiak/Development/dts/open-source-firmware-validation/log.html
Report:  /home/mkusiak/Development/dts/open-source-firmware-validation/report.html
```

</p>
</details> 

Will try qemu next

3mkusiak commented on docs/zarhus-tests.md:
```diff
@@ -0,0 +1,260 @@
+    -v dts_ipxe_link:https://boot.dasharo.com/dts/dts-rc.ipxe \
+    -v features:encryption,otab \
+    -v zarhus_wic_gz_file:../zarhus-base-image-genericx86-64.rootfs.wic.gz \
+    -v target_device:nvme0n1 -t "ZHS001*" zarhus-x86/zarhus.robot
```
Let's run all tests by default.

m-iwanicki replied:
[82184..d10cc](https://github.com/Dasharo/open-source-firmware-validation/compare/821849f25a1e6248235c70942eb7516f4f323a18..d10cc6344d54e1cd8c1e759af00b9206e860632b)

ghost commented:
Rerun full test set this time, all tests do pass. 

<details><summary>ZarhusOS full scope HW</summary>
<p>

```log
==============================================================================
Zarhus                                                                        
==============================================================================
ZHS001.206 Make sure that cukinia tests pass                          | PASS |
------------------------------------------------------------------------------
ZHS002.206 Make sure that update works :: Make sure that update vi... | PASS |
------------------------------------------------------------------------------
ZHS003.206 Make sure that failed update rollbacks to working slot ... | PASS |
------------------------------------------------------------------------------
ZHS004.206 Make sure that rollback works :: Make sure that `otab r... | PASS |
------------------------------------------------------------------------------
Zarhus                                                                | PASS |
4 tests, 4 passed, 0 failed
==============================================================================
Debug:   /home/mkusiak/Development/dts/open-source-firmware-validation/command_log.
Output:  /home/mkusiak/Development/dts/open-source-firmware-validation/output.xml
Log:     /home/mkusiak/Development/dts/open-source-firmware-validation/log.html
Report:  /home/mkusiak/Development/dts/open-source-firmware-validation/report.html
```

</p>
</details> 

Skipping doing tests on QEMU, waste of time if hardware works.

ghost commented:
Rebased on top of develop. Waiting for checks to complete.
